src/app/app.datatypes.ts

```diff
@@ -63,14 +64,32 @@ export class GetAddressResponseTransaction {
   txid: string;
 }
 
-export function parseGetAddressTransaction(raw: GetAddressResponseTransaction): Transaction {
+export function parseGetAddressTransaction(raw: GetAddressResponseTransaction, address: string): Transaction {
+
+  //Detect if the address sent or received the coins.
+  let incoming: boolean = true;
+  if (raw.inputs[0].owner.toLowerCase() == address.toLowerCase()) { incoming = false; }
+
+  //Calculate the amount of incoming or outgoing coins, as appropriate.
+  let coins:number = 0;
```

Comment:
coins does not need to be cast as a number, will automatically be cast due to default value being a number